osu.Game.Rulesets.Catch/Objects/Drawables/DrawableCatchHitObject.cs

```diff
@@ -20,12 +21,19 @@ public abstract class DrawableCatchHitObject : DrawableHitObject<CatchHitObject>
 
         protected override float SamplePlaybackPosition => HitObject.X / CatchPlayfield.WIDTH;
 
+        public int RandomSeed => HitObject?.RandomSeed ?? 0;
+
         protected DrawableCatchHitObject([CanBeNull] CatchHitObject hitObject)
             : base(hitObject)
         {
             Anchor = Anchor.BottomLeft;
         }
 
+        /// <summary>
+        /// Get a random number in range [0,1) based on seed <see cref="RandomSeed"/>.
+        /// </summary>
+        public float RandomSingle(int series) => StatelessRNG.NextSingle(RandomSeed, series);
```

Comment:
Is there a particular reason why this and `RandomSeed` are public here? Would rather prefer `protected float RandomSingle(int)` and `private int randomSeed` if at all possible.

Reply:
I originally wanted to use it here <https://github.com/ppy/osu/blob/a16b265090942f7c5fa06234978005776586ff5b/osu.Game.Rulesets.Catch/UI/Catcher.cs#L211-L213> (not included in this PR).
I also imagine a use case where mod want to apply custom random transformation to a hit object.